contributors/devel/sig-testing/testing-strategy.md

```diff
@@ -0,0 +1,99 @@
+## Defining a Robust Testing Strategy
+
+This document outlines a testing strategy for Kubernetes features based on the **testing pyramid**, with considerations on the existing CI system's characteristics.
```

Comment:
@aojea let's add the diagram from the talk?

Reply:
> It also implies that E2E tests are built on top of integration tests and thus depend on them. 

there is no dependency between layers, let me add the visualization

Reply:
added a link to martin fowlers https://martinfowler.com/articles/practical-test-pyramid.html, it is a metaphore.
We presented it in Detroit Kubecon https://sched.co/182Oe and I presented in some sig testing meetings https://docs.google.com/presentation/d/1gjZMMr_sjLkPlF4a4p5qrvFpCPlU2NWATMGA27tXxxo/edit#slide=id.p

We are implementing the pyramid of testing in kubernetes, so this is not a declaration of intentions or a proposal to change things, is stating what we are doing ... I have a long record of probed success with this method in multiple projects that were a disaster of quality, said this, every feature owner is free to implement the strategy they want if they disagree with this

Reply:
https://martinfowler.com/articles/practical-test-pyramid.html :
> Unfortunately the concept of the test pyramid falls a little short if you take a closer look. 

Looks like others hard similar concerns about this metaphor. But I'm fine with using it, now that it gets introduces so that readers can follow up on what it means.

Reply:
Let me rephrase, I think now I understand what you mean, and I agree these metaphors are guidance not rules, and each project or feature is different, so some cases it does not apply at all

Reply:
please see last commit to see if that matches better your expectactions

Reply:
Yes, looks good.